### PR TITLE
Replace uniVocity CSV library with jackson-csv #702

### PR DESCRIPTION
**Pull request by rkampani**
# <Feature Title>

## Fixes Issue
- [https://github.com/authorjapps/zerocode/issues/702] Which issue or ticket was(will be) fixed by this PR? (capture the issue number)

PR Branch
**https://github.com/rkampani/zerocode/tree/feature/701-replaceUniVocityJar **

## Checklist:

* [x] New Unit tests were added
  * [ ] Covered in existing Unit tests

* [ ] Integration tests were added
  * [ ] Covered in existing Integration tests

* [x] Test names are meaningful

* [ ] Feature manually tested and outcome is successful

* [x] PR doesn't break any of the earlier features for end users
  * [ ] WARNING! This might break one or more earlier earlier features, hence left a comment tagging all reviewrs

* [x] Branch build passed in CI

* [x] No 'package.*' in the imports

* [ ] Relevant DOcumentation page added or updated with clear instructions and examples for the end user
  * [ ] Not applicable. This was only a code refactor change, no functional or behaviourial changes were introduced

* [ ] Http test added to `http-testing` module(if applicable) ?
  * [ ] Not applicable. The changes did not affect HTTP automation flow

* [ ] Kafka test added to `kafka-testing` module(if applicable) ?
  * [ ] Not applicable. The changes did not affect Kafka automation flow

### CODE REVIEW

**Comment by authorjapps** [comment]
@rkampani , thanks for the PR. Assigned the reviewers too.
The CI build is in-progress [here](https://github.com/authorjapps/zerocode/pull/711/checks).
Once Green, we should be able to merge this.

**Review by javiertuya — CHANGES_REQUESTED** [review]
Thanks @rkampani, great job. It looks good to me.
- I've placed a comment about empty lines in the code.
- Also, I realized that csv parsing was not unit tested (existing tests for Parametrized and the DB SQL Executor do not consider low level edge cases) @authorjapps. So that, I prepared a few tests that run in the main branch that detect a couple of discrepancies when they run using the migrated parser. These tests are reproduced below. Please, feel free to use and/or adapt these tests.

```
package org.jsmart.zerocode.core.di;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertNull;

import org.jsmart.zerocode.core.di.main.ApplicationMainModule;
import org.jukito.JukitoRunner;
import org.jukito.TestModule;
import org.junit.Test;
import org.junit.runner.RunWith;

import com.google.inject.Inject;
import com.univocity.parsers.csv.CsvParser;

@RunWith(JukitoRunner.class)
public class CsvParserTest {

    public static class JukitoModule extends TestModule {
        @Override
        protected void configureTest() {
            ApplicationMainModule applicationMainModule = new ApplicationMainModule("config_hosts_test.properties");
            install(applicationMainModule);
        }
    }

    @Inject
	CsvParser parser;

	@Test
	public void testCsvparseSpaces() {
		assertThat(parser.parseLine(" abc ,\t de f  , ghi "), is(new String[] { "abc", "de f", "ghi" }));
	}

	@Test
	public void testCsvParseEmptyCell() {
		assertThat(parser.parseLine(",,  , \t ,"), is(new String[] { null, null, null, null, null }));
	}

	@Test
	public void testCsvparseEmptyLine() {
		assertNull(parser.parseLine(""));
		assertThat(parser.parseLine("  "), is(new String[] { null }));
		assertThat(parser.parseLine(" \t "), is(new String[] { null }));
	}

	@Test
	public void testCsvParseQuotesAndOtherChars() {
		assertThat(parser.parseLine("a'c, d\"f, x\\y"), is(new String[] { "a'c", "d\"f", "x\\y" }));
		assertThat(parser.parseLine("euro\u20AC, año, naïf"), is(new String[] { "euro\u20AC", "año", "naïf" }));
	}

	@Test
	public void testCsvParseWindowsEndings() { // assume that lines where already splitted by \n
		assertThat(parser.parseLine("abc, def\r"), is(new String[] { "abc", "def" }));
		assertThat(parser.parseLine("abc, def \r"), is(new String[] { "abc", "def" }));
		// empty lines
		assertThat(parser.parseLine("\r"), is(new String[] { null })); // should be null?, see testCsvparseEmptyLine
		assertThat(parser.parseLine("  \r"), is(new String[] { null }));
		assertThat(parser.parseLine(" \t \r"), is(new String[] { null }));
	}

}
```

**Comment by rkampani** [comment]
Thank you for the input and findings @javiertuya . I’ll incorporate those changes and appreciate for providing the unit tests as well. 🙂 I’ll make sure those are addressed in the next commit.

File: core/src/main/java/org/jsmart/zerocode/core/db/DbCsvLoader.java [hunk 1]
```diff
@@ -10,20 +10,21 @@
 
 import org.apache.commons.dbutils.QueryRunner;
 import org.apache.commons.lang3.StringUtils;
+import org.jsmart.zerocode.core.di.provider.CsvParserProvider;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.univocity.parsers.csv.CsvParser;
+
```
[the remark this answers is not in the document]
Comment (rkampani) [reply core/src/main/java/org/jsmart/zerocode/core/db/DbCsvLoader.java:17]: All the changes have been incorporated. Please re-review

**Review by javiertuya — COMMENTED** [review]
Thanks @rkampani. I think that it is ready.

@authorjapps Before merging, I would like to ask a question:

As part as this PR the the `univocity-parser` is replaced by `jackson-dataformat-csv` and also, the `gson` dependency is removed from core.

Looking at the parent pom, `gson`  is declared as v2.10.1. Looking at the dependency:tree report, `gson` v2.10.1 is introduced as a transitive dependency by `com.aventstack:extentreports`.

Should we keep the `gson` dependency, or remove it?

**Comment by nirmalchandra** [comment]
## Next >>
@rkampani, Do you know why you removed "gson" from core? 

I see `KafkaSender` and many other Kafka features uses Gson. 
Example: 
```
ProducerRecord record = gson.fromJson(line, ProducerRecord.class);
```

So I think, we can't get rid of it now at this point of time from "core", or should not be dependent on `extentreports` to pull this "gson" transitively. The version "2.10.1" is matching because of Parent POM still has it, otherwise it might pull a different version(as per Extentreport module) 
 
So let's pull it via the "core" module explicitly in compile scope as it was doing earlier.
I would suggest not to remove it 🔴  and put the "gson" back to the "core" module.

Let me know if you think otherwise :)

File: core/pom.xml [hunk 2]
```diff
@@ -63,12 +63,8 @@
             <artifactId>jackson-dataformat-yaml</artifactId>
         </dependency>
         <dependency>
-            <groupId>com.univocity</groupId>
-            <artifactId>univocity-parsers</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>com.google.code.gson</groupId>
-            <artifactId>gson</artifactId>
```
Comment (nirmalchandra) [inline core/pom.xml:71]: I would suggest not to remove it 🔴 and put the "gson" back to the "core" module.

_(as explained in the PR discussion thread)_

Comment (rkampani) [reply core/pom.xml:71]: I believe the removal from core was unintentional—apologies for the oversight. Since the Kafka integration depends on it, we definitely need to keep it. The tests likely passed because the jar is still being pulled in through the parent [pom.xml ](https://github.com/rkampani/zerocode/blob/565f45e574fd53f1e3429bf459b2f8769d50d351/pom.xml#L137), which is probably why we didn’t see any immediate test failures . I’ll push the commit later today to restore it, then run a full build along with the requested report to ensure everything is functioning as expected.

**Comment by rkampani** [comment]
Here are screenshot reports for reports generation 

Screenshot 2025-04-29 at 8 06 53 PM
Screenshot 2025-04-29 at 8 07 19 PM
Screenshot 2025-04-29 at 8 07 31 PM

**Review by javiertuya — APPROVED** [review]
After the latest changes in the dependencies specified in the pom, it is ready to merge.
Now updating the branch

**Comment by nirmalchandra** [comment]
Attached the following for comparison purpose only and future reference. 
No action is needed.

[mvntree0_removed_both.txt - Removed from Both](https://github.com/user-attachments/files/19998718/mvntree0_removed_both.txt)

[mvntree1_removed_core.txt - Removed form Core only](https://github.com/user-attachments/files/19998719/mvntree1_removed_core.txt)

[mvntree2_not_removed.txt - Not Removed](https://github.com/user-attachments/files/19998721/mvntree2_not_removed.txt)
